tests/integration/test_annotate_image.py

```diff
@@ -0,0 +1,202 @@
+                    "children": ["en:eu-organic"],
+                },
+                "en:eu-organic": {
+                    "wikidata": {"en": "Q380448"},
+                    "parents": ["en:organic"],
+                },
+            }
+        return robotoff.taxonomy.Taxonomy.from_dict(data) if data else None
+
+    monkeypatch.setattr(
+        robotoff.taxonomy,
+        "fetch_taxonomy",
+        fetch_taxonomy_mock,
+    )
+
+
+_AUTH_HEADER = {"Authorization": "Basic " + base64.b64encode(b"a:b").decode("ascii")}
+
+
+def test_image_brand_annotation(client, monkeypatch, fake_taxonomy):
+    ann = LogoAnnotationFactory(image_prediction__image__source_image="/images/2.jpg")
+    barcode = ann.image_prediction.image.barcode
+    _fake_store(monkeypatch, barcode)
+    monkeypatch.setattr(
+        BRAND_PREFIX_STORE, "get", lambda: {("Etorki", "0000000xxxxxx")}
+    )
+    start = datetime.utcnow()
+    result = client.simulate_post(
+        "/api/v1/images/logos/annotate",
+        json={
+            "withCredentials": True,
+            "annotations": [{"logo_id": ann.id, "value": "etorki", "type": "brand"}],
+        },
+        headers=_AUTH_HEADER,
+    )
+    end = datetime.utcnow()
+    assert result.status_code == 200
+    assert result.json == {"created insights": 1}
+    ann = LogoAnnotation.get(LogoAnnotation.id == ann.id)
+    assert ann.annotation_type == "brand"
+    assert ann.annotation_value == "etorki"
+    assert ann.annotation_value_tag == "etorki"
+    assert ann.taxonomy_value == "Etorki"
+    assert ann.username == "a"
+    assert start <= ann.completed_at <= end
+    # we generate a prediction
+    predictions = list(Prediction.select().filter(barcode=barcode).execute())
+    assert len(predictions) == 1
+    (prediction,) = predictions
+    assert prediction.type == "brand"
+    assert prediction.data == {
+        "logo_id": ann.id,
+        "confidence": 1.0,
+        "username": "a",
+        "is_annotation": True,
+        "notify": True,
+    }
+    assert prediction.value == "Etorki"
+    assert prediction.value_tag == "Etorki"
+    assert prediction.predictor == "universal-logo-detector"
+    assert start <= prediction.timestamp <= end
+    assert prediction.automatic_processing
+    # that in turn generated an insight
+    insights = list(ProductInsight.select().filter(barcode=barcode).execute())
+    assert len(insights) == 1
+    (insight,) = insights
+    assert insight.type == "brand"
+    assert insight.data == {
+        "logo_id": ann.id,
+        "confidence": 1.0,
+        "username": "a",
+        "is_annotation": True,
+        "notify": True,
+    }
+    assert insight.value == "Etorki"
+    assert insight.value_tag == "Etorki"
+    assert insight.predictor == "universal-logo-detector"
+    assert start <= prediction.timestamp <= end
+    assert insight.automatic_processing
+    assert insight.username == "a"
+    assert insight.completed_at is None  # we did not run annotate yet
+
+
```

Comment:
```suggestion
```
This test will check that, given an image with a logo above the confidence threshold, that is then fed into the ANN logos and labels model, we annotate properly a product.
```
```